### ISSUE TITLE
Version bump `0.2.11` wishlist

### ISSUE THREAD

**Issue by VuiMuich** [leftwm/leftwm#598]
This issue is going to be used as a wishlist for issues to be fixed for the `0.2.11` release.

Also feel free to briefly suggest wishes or ideas not yet tracked by an issue and eventually open the issue later on.

**Comment by di-effe** [comment]
A function to retrieve all key bindings like Awesome do would be nice

**Comment by mautamu** [comment]
> A function to retrieve all key bindings like Awesome do would be nice

In leftwm-state?

**Comment by di-effe** [comment]
> 

I guess that would work too, parsing the result with a liquid template and passing the result to something like Rofi.
But tbh I was thinking at something that works out of the box, displaying a popup/overlay on screen

**Comment by VuiMuich** [comment]
> > 
> 
> 
> 
> I guess that would work too, parsing the json with a liquid template and passing the result to something like Rofi.
> 
> But tbh I was thinking at something that works out of the box, displaying a popup/overlay on screen

I think this would be a nice feature for `lefthk` actually, but also only printing the keybinds. Putting this in a nice window definitely should be a task for conky or eww, just to name two possibilities.
Adding this directly in leftwm/lefthk could easily be the starting point of bloating things.

**Comment by di-effe** [comment]
I get your point and make sense.

**Comment by di-effe** [comment]
I am not sure if this would be technically possible or it would make any sense or use, but (<- magic word) what about managing the 4 window borders (optionally) separately? In theory it could open new possibilities (and nightmares for the eyes) for theming.

For example having a bottom border thicker and/or with a different accent color.

**Comment by blackdragon2447** [comment]
the ability to set keybinds in the `theme.toml`

**Comment by VuiMuich** [comment]
> the ability to set keybinds in the `theme.toml

I see your desire, though I think this would be better kept as a 'strong recommendation to use these keybinds' in the README of the theme, as I don't want the theme to interfere with my workflow. I think this is exactly the beauty of leftwm's concept that the theme is widely agnostic of the users workflow.
Maybe if we introduce a `block_theme_overrides = true`  in the config...

**Comment by VuiMuich** [comment]
> I am not sure if this would be technically possible or it would make any sense or use, but (<- magic word) what about managing the 4 window borders (optionally) separately? In theory it could open new possibilities (and nightmares for the eyes) for theming.
> 
> For example having a bottom border thicker and/or with a different accent color.

 If technically possible (have not checked yet, how borders get drawn exactly) this would definitely be interesting and would mix well with #369 which is currently on the road map for 0.2.11 already. I imagine especially double boarders only on a single edge might actually be a visually very pleasing indicator. It might even help to indicate the snap location, for `snap_to_tile` etc.

**Comment by muemmel** [comment]
Cheers. First off some praise:
Been using leftwm for a few weeks now and am really happy with it. Apart from some issues that seemingly will get attention in the future (window rules and floating fixes, specifically) I really enjoy the simplicity of it. It's great to work with and is easy to set up for my 32:9 screen.

I'm mentioning my screens aspect ratio, because I would love to see two features that would make it even more usable:

1:
Please allow for an option (or perhaps another Layout) for the CenterMain(Balanced) Layout to reserve the unoccupied with only two windows opened.
To elaborate: When using the layout with a single window, that window is centered. Great. Opening another window, however, splits the screen space evenly. One window left, one right, according to max. width, padding and gutters. Opening another window applies the actual layout. One to the left, Main in center, one to the right.
I would really like having the 'actual' layout applied with only two windows open, w/o having to open a literal placeholder terminal.

2:
An additional layout similar to MainAndDeck. Essentially like CenterMain, except one card stack on either side, possibly with separate key bindings for each deck.

Again, great WM. I'm even considering learning Rust so I can contribute.

Thank you for your hard work and keep it up! :)

**Comment by VuiMuich** [comment]
> 2:
> 
> An additional layout similar to MainAndDeck. Essentially like CenterMain, except one card stack on either side, possibly with separate key bindings for each deck.

I think @hertg's layout rework might open some possibilities for things like this. 

> Again, great WM. I'm even considering learning Rust so I can contribute.

Learning Rust is always a great choice imo.

**Comment by rigoletto-freebsd** [comment]
Please, consider implementing  #343.

**Comment by federicotorrielli** [comment]
Hi, what do you think about #632 ?

**Comment by godalming123** [comment]
Some code that can be called when A focusable window loses focus. In my usecase I have a menu window and I want it to hide when I click on the wallpaper or focus an app.

**Comment by VuiMuich** [comment]
> Some code that can be called when A focusable window loses focus. In my usecase I have a menu window and I want it to hide when I click on the wallpaper or focus an app.

So you basically wish for a `window_rule` that goes like `hide_on_unfocus = true`? Thats a nice idea.

**Comment by godalming123** [comment]
> > Some code that can be called when A focusable window loses focus. In my usecase I have a menu window and I want it to hide when I click on the wallpaper or focus an app.
> 
> So you basically wish for a `window_rule` that goes like `hide_on_unfocus = true`? Thats a nice idea.

Yes something like that but possibly instead close as it's more accurate to the defenition and maybe if it's easy to implement some kind of custom function so you could do whatever you want like send a notification on focus loss

**Comment by VuiMuich** [comment]
`execute_on_unfocus` sounds like a useful idea.

**Comment by AethanFoot** [comment]
> Everyone has had some really good ideas for 3.0.0 and done amazing work! I am always truly amazed by what the community produces. Historically I have release LeftWM no less then every three months. My first thought when we all started talking 3.0.0 was to be a little patient, and I can still be if everyone wants to continue to hold off the release. But at the same time, there will always be just one more thing...
> 
> How does everyone feel about doing one final stretch if anyone want something added and then shoot for a release early next week? Just throwing out Ideas here. There is just so much good stuff in the 3.0.0 branch I am itching to share it with the world.

I 100% agree, there are a couple issues that ideally would be solved (I may have time to solve them). One is jetbrains applications not taking focus (they really are a pain set applications :laughing:) and cant remember the others right now. I was wanting to do more this release but final year of uni has taken priority (but will nicely be done soon). It is really nice as well to see some new names contributing.

Thanks!

**Comment by mdupuis13** [comment]
> I am not sure if this would be technically possible or it would make any sense or use, but (<- magic word) what about managing the 4 window borders (optionally) separately? In theory it could open new possibilities (and nightmares for the eyes) for theming.
> 
> For example having a bottom border thicker and/or with a different accent color.

I feel a Star Trek theme coming !
